### ISSUE TITLE
[displayName] questioning if its still needed

### ISSUE THREAD

**Issue by Andarist** [facebook/react#10060]
When its used:
- error messages
- dev tools
- HOCs

I would argue that its valuable only in `DEV` mode. Even when it comes to error messages I think I've never seen any production log containing it, I cannot be sure though if it never is.

And if its valuable only in DEV mode, shouldnt we just ditch adding them (or write a ditching transform/codemod) as they can be inferred from the functions/classes/variables anyway? The only remaining use case would be using `displayName` for HOCs.

If Im wrong let me know. This would however fix 2 minor problems:
- smaller build outputs (slightly)
- remove problem of unintentionally preventing tree-shaking of SFCs in flat bundles (more can be read [here](https://iamakulov.com/notes/polished-webpack/))

**Comment by Andarist** [comment]
Thanks for a quick answer, it was a habit I've developed and didnt quite question it till today.

Also got impression that its added in the React's docs - which is not true. However some `displayName`s are still floating around, i.e. [here](https://github.com/reactjs/react-redux/blob/5939d906e1281ac0241d4d8e7a8eed9a1150424e/src/components/Provider.js#L56).

**Comment by gaearon** [comment]
It’s not really a big deal in React Redux, there's like two components there 😛 . For libraries with 100s of components I would definitely not recommend this pattern but in this case it hardly matters.

**Comment by Andarist** [comment]
Yeah, it surely doesnt matter code-wise, but keeping this 'legacy' in such popular library kinda gives a wrong example and somebody might use it without real need in other project
